apps/designer/app/canvas/shared/instance.ts

```diff
@@ -194,24 +194,27 @@ export const usePublishSelectedInstanceData = (treeId: Tree["id"]) => {
   const [allUserProps] = useAllUserProps();
 
   useEffect(() => {
-    // Unselects the instance by `undefined`
-    let payload;
-    if (instance !== undefined) {
-      const props =
-        allUserProps[instance.id] ??
-        utils.props.createInstanceProps({ instanceId: instance.id, treeId });
-      const browserStyle = getBrowserStyle(selectedElement);
-      payload = {
-        id: instance.id,
-        component: instance.component,
-        cssRules: instance.cssRules,
-        browserStyle,
-        props,
-      };
-    }
-    publish({
-      type: "selectInstance",
-      payload,
+    // some styles require nexxxt frame to provide new value
+    requestAnimationFrame(() => {
```

Comment:
I think you might be fixing it in the wrong place. The core issue is that this logic doesn't depend on when the style updates were actually applied, we are probably dispatching the new instance before we actually set the styles, can you please verify this?